src/main/java/utilities/CalendarManager.java

```diff
@@ -0,0 +1,173 @@
+package utilities;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Calendar;
+import java.util.Date;
+import java.text.DateFormat;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class CalendarManager {
+    private String dateActualComplete;
+    private Date date;
+    private Calendar calendarDate;
+    private int dates;
+    private DateFormatInput dateFormat;
+
+    public CalendarManager() {
+    }
+
+    /**
+     * Converts string to date according input and format of date.
+     *
+     * @param input is value which will convert to date
+     * @return object type Date
+     */
+    public Date convertStringToDate(final String input) throws ParseException {
+        if (input == null | input == "") {
+            throw new RuntimeException("Invalid value or date");
+        }
+        String[] analise = input.split(" ");
+        generateDateActual();
+        if (analise.length > 1) {
+            isRightDateAccordingConventions(input);
+            managerInputs();
+            return getResult();
+        } else {
+            if (isValidateFormatAndDate(input)) {
+                return resultRightFormat(input);
+            }
+            onlyWordInput(input);
+            managerInputs();
+            return getResult();
+        }
+    }
+
+    /**
+     * Convert a string to data.
+     * This convert string which contains a int and string according
+     * of enum DateFormatInput.
+     *
+     * @param input is value which will convert to date
+     */
+    private void isRightDateAccordingConventions(final String input) {
+        String[] divide = input.toLowerCase().split(" ", 2);
+        dates = Integer.parseInt(divide[0]);
+        String fillUpLowBar = divide[1].replace(' ', '_');
+        dateFormat = DateFormatInput.valueOf(fillUpLowBar);
+        if (dateFormat.isSingular() && dates != 1) {
+            throw new RuntimeException("The value is incorrectly entered or has negative number");
+        }
+        if (!dateFormat.isSingular() && dates == 1) {
```

Comment:
The is singular condition is repeated. Maybe it could be refactored.

Reply:
Solved I change the message to get meaning exception